radix-engine/tests/determinism.rs

```diff
@@ -17,3 +24,58 @@ fn test_simple_deterministic_execution() {
         .substate_store()
         .assert_eq(test_runner1.substate_store());
```

Comment:
Should we add an assert here that they have equal state hash?

Reply:
I can add it if you wish.
A better question (which popped up to me after seeing this line earlier) is: why do we need a test util for comparing state hash, if we have a test util for comparing exactly the entire state? (it seems like even better way of asserting determinism, and the performance is probably acceptable, given typical test state sizes)

Reply:
Haha I didn't actually know we had this(!). I'm surprised that SubstateStore has equality defined on it!

But yeah, good point :p